images/newrelic/configs/main.tf

```diff
@@ -0,0 +1,59 @@
+terraform {
+  required_providers {
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+locals {
+  components = {
+    "fluent-bit-output" = {
+      packages = ["newrelic-fluent-bit-output${var.suffix}", "newrelic-fluent-bit-output-compat"]
+    },
+    "infrastructure-bundle" = {
+      packages = ["newrelic-infrastructure-bundle${var.suffix}", "newrelic-infrastructure-agent"]
+    },
+    "k8s-events-forwarder" = {
+      packages = ["newrelic-infrastructure-agent${var.suffix}"]
+    },
+    "kube-events" = {
+      packages = ["newrelic-nri-kube-events${var.suffix}"]
+    },
+    "kubernetes" = {
+      packages = ["nri-kubernetes${var.suffix}"]
+    },
+    "prometheus-configurator" = {
+      packages = ["newrelic-prometheus-configurator${var.suffix}"]
+    },
+    "prometheus" = {
+      packages = ["nri-prometheus${var.suffix}"]
+    },
+  }
+}
+
+variable "name" {
+  description = "Package name (e.g. server or repo-server)"
+}
+
+variable "suffix" {
+  description = "Package name suffix (e.g. version stream)"
+  default     = ""
+}
+
+variable "extra_packages" {
+  description = "Additional packages to install."
+  type        = list(string)
+  default     = []
+}
+
+data "apko_config" "this" {
+  config_contents = file("${path.module}/template.${var.name}.apko.yaml")
+  extra_packages  = concat(local.components[var.name].packages, var.extra_packages)
+}
+
+output "config" {
+  value = jsonencode(data.apko_config.this.config)
+}
+
+output "main_package" {
+  value = local.components[var.name].packages[0]
+}
```

Comment:
I like the idea here, but I wonder if we could make the `[0]` more explicit by adding more structure to the types above?  e.g.

```hcl
  components = {
    "fluent-bit-output" = {
      main_package = "newrelic-fluent-bit-output${var.suffix}"
      extra_packages = ["newrelic-fluent-bit-output-compat"]
    },
   ...
```

Reply:
ah that's a cool idea, I wasn't happy about that, thanks!